extensions/wikia/EmergencyBroadcastSystem/EmergencyBroadcastSystemController.class.php

```diff
@@ -19,8 +24,21 @@ public static function saveUserResponse( $val ) {
 			$wgUser->setOption( $option_name, $timestamp );
 		}
 	}
+	// PROTECTED
+
+	protected function isCorrectPage() {
+		$title = $this->getContext()->getTitle();
+		$specialPageName = Transaction::getAttribute(Transaction::PARAM_SPECIAL_PAGE_NAME);
+
+		return $title->isContentPage() || $specialPageName === 'WikiActivity' || $specialPageName === 'Recentchanges';
+	}
+
+	protected function userIsPowerUser() {
+		$user = $this->getContext()->getUser();
+		return $user->isPowerUser();
+	}
 
-	public static function canOpenEBS( ) {
+	protected function canOpenEBS( ) {
```

Comment:
I always have problem with this... If something could be static why not to keep it as such - could be used in more places then - specifically I will need it to decide about adding js.